### PR TITLE
Support repeatable read and serializable isolation level

### PR DESCRIPTION
**Pull request by tanscorpio7**
### Description

Currently we do not support repeatable read and serializable isolation levels. 
With this PR 
BABEL repeatable read &#8594;  PG repeatable read (if repeatable read escape hatch is set to pg_isolation)
BABEL serializable &#8594; PG serializable (if serializable escape hatch is set to pg_isolation)

REPEATABLE READ GUC:- 'repeatable_read_isolation'
SERIALIZABLE GUC:- 'serializable_isolation'

### Issues Resolved

BABEL-4146 & BABEL-4145

### Test Scenarios Covered ###
JDBC : Setting the GUC/ setting all different isolation level. 
Python : Concurrent transaction run with one of them being repeatable read while other is read committed. Steps like select, update, delete, insert are used. Different permutations are run to force read/write, write/write concurrency.  

Signed-off-by: Tanzeel Khan <tzlkhan@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 1]
```diff
@@ -4221,11 +4221,18 @@ tsql_IsolationLevelStr:
 				}
 			| REPEATABLE READ
 				{
-					TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
-					ereport(ERROR,
+					if(pltsql_enable_snapshot_isolation_for_reapeatable_read == false){
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: let us switch if/else for better readability

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 2]
```diff
@@ -4221,11 +4221,18 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
+						ereport(ERROR,
 							(errcode(ERRCODE_SYNTAX_ERROR),
-							errmsg("REPEATABLE READ isolation level is not supported"),
+							errmsg("REPEATABLE READ isolation level is not supported, consider setting babelfishpg_tsql.enable_snapshot_isolation_for_reapeatable_read to on to enable snapshot isolation instead"),
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: this will require DBE approval

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 3]
```diff
@@ -4221,11 +4221,18 @@ tsql_IsolationLevelStr:
...
 							parser_errposition(@1)));
+					}
+					else{
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_SNAPSHOT);
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: it should be INST_TSQL_ISOLATION_LEVEL_REPEATABLE_READ in both if/else

File: contrib/babelfishpg_tsql/src/guc.c [hunk 4]
```diff
@@ -1148,6 +1149,15 @@ define_custom_variables(void)
 							 PGC_SUSET,
 							 GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE | GUC_DISALLOW_IN_AUTO_FILE,
 							 NULL, NULL, NULL);
+
+	DefineCustomBoolVariable("babelfishpg_tsql.enable_snapshot_isolation_for_reapeatable_read",
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c]: need DBE discussion on this, we might also have a escape hatch instead of GUC

File: test/JDBC/expected/TestIsolationLevels.out [hunk 5]
```diff
@@ -8,7 +8,7 @@ set transaction isolation level repeatable read;
 go
 ~~ERROR (Code: 33557097)~~
 
-~~ERROR (Message: REPEATABLE READ isolation level is not supported)~~
```
Comment (surendravishnoi) [inline test/JDBC/expected/TestIsolationLevels.out:11]: we should also add test cases where we set GUC to true and repeatable read starts working. We should test GUC setting at both database/session level

File: test/python/input/isolation/BABEL-4146.spec [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+setup		{  select set_config('babelfishpg_tsql.enable_snapshot_isolation_for_reapeatable_read','on',false);
+  set transaction isolation level repeatable read; BEGIN TRAN; SET lock_timeout '500'; }
+step s1t { Select current_setting('transaction_isolation'); }
+step s1s { Select * from child; }
+step s1i	{ INSERT INTO child VALUES (1, 1); }
+step s1c	{ COMMIT; }
+
+session s2
+setup		{ BEGIN TRAN; SET lock_timeout '10000'; }
+step s2i	{ INSERT INTO child VALUES (2, 1); }
```
Comment (surendravishnoi) [inline test/python/input/isolation/BABEL-4146.spec:35]: we want to add update/delete as well

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 7]
```diff
@@ -4221,11 +4221,18 @@ tsql_IsolationLevelStr:
 				}
 			| REPEATABLE READ
 				{
-					TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
-					ereport(ERROR,
+					if(pltsql_enable_repeatable_read_and_serialization == true){
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: == true is not required, if (x) should be sufficient for boolean types

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 8]
```diff
@@ -4221,11 +4221,18 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
+						$$ = "repeatable read";	
+					}
+					else{
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: this should be INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 9]
```diff
@@ -4234,11 +4241,17 @@ tsql_IsolationLevelStr:
 				}
 			| SERIALIZABLE
 				{
-					TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
-					ereport(ERROR,
+					if(pltsql_enable_repeatable_read_and_serialization == true){
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: if (x) for booleans

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 10]
```diff
@@ -4234,11 +4241,17 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
+						$$ = "serializable";
+					}
+					else{
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: should be INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_SERIALIZABLE

File: contrib/babelfishpg_tsql/src/guc.c [hunk 11]
```diff
@@ -66,6 +66,7 @@ char	   *pltsql_host_release = NULL;
 char	   *pltsql_host_service_pack_level = NULL;
 
 bool		pltsql_enable_create_alter_view_from_pg = false;
+bool		pltsql_enable_repeatable_read_and_serialization = false;
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c]: why are we using single GUC for both repeatable read and serialization, we should use two escape hatches

File: contrib/babelfishpg_tsql/src/guc.c [hunk 12]
```diff
@@ -1148,6 +1149,15 @@ define_custom_variables(void)
 							 PGC_SUSET,
 							 GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE | GUC_DISALLOW_IN_AUTO_FILE,
 							 NULL, NULL, NULL);
+
+	DefineCustomBoolVariable("babelfishpg_tsql.enable_repeatable_read_and_serialization",
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c]: Let us have two GUCs, enable_repeatable_read_isolation_level and enable_serializabl_isolation_level

File: test/JDBC/expected/BABEL-3214.out [hunk 13]
```diff
@@ -98,3 +98,110 @@ smallint
 5
 ~~END~~
 
+
+SELECT SET_CONFIG('babelfishpg_tsql.enable_repeatable_read_and_serialization','on',false);
```
Comment (surendravishnoi) [inline test/JDBC/expected/BABEL-3214.out]: let us have test case where we set GUCs at DB level, create a new connection and set both RR and SR

File: test/python/input/isolation/BABEL-4146.spec [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+teardown
+{
+  DROP TABLE parent, child;
+}
+
+session s1
+setup     { SELECT set_config('babelfishpg_tsql.enable_repeatable_read_and_serialization','on',false);
+  set transaction isolation level repeatable read; BEGIN TRAN; SET lock_timeout '500'; }
```
Comment (surendravishnoi) [inline test/python/input/isolation/BABEL-4146.spec:25]: we have not added anything for serializable?

File: test/python/input/isolation/BABEL-4146.spec [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+permutation s1i s1c s2i s2c
+permutation s1i s2i s1c s2c
+permutation s1i s2i s1c s2c
+permutation s1i s2i s1c s2c
+permutation s1i s2i s1c s2c
+permutation s1i s2i s2c s1c
+permutation s1i s2i s2c s1c
+permutation s2i s1i s1c s2c
+permutation s2i s1i s1c s2c
+permutation s2i s1i s2c s1c
```
Comment (surendravishnoi) [inline test/python/input/isolation/BABEL-4146.spec]: why these duplicates?

File: test/python/expected/pyodbc/BABEL-4146.out [hunk 16]
```diff
@@ -0,0 +1,286 @@
...
+
+step s2c: COMMIT;
+
+starting permutation : { s1u s2u s2c s1c }
+step s1u: UPDATE parent SET aux = 'bar';
+~~ROW COUNT: 1~~
+
+step s2u: UPDATE parent SET aux = 'baz'; <waiting ...>
+~~ERROR (Code: 33557097)~~
+~~ERROR (Message: [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]canceling statement due to lock timeout (33557097) (SQLExecDirectW))~~
```
Comment (surendravishnoi) [inline test/python/expected/pyodbc/BABEL-4146.out]: how much is the timeout value?

Comment (tanscorpio7) [reply test/python/expected/pyodbc/BABEL-4146.out]: 10sec, will remove this permuation

File: test/python/input/isolation/BABEL-4146.spec (hunk 14 above)
Comment (tanscorpio7) [reply test/python/input/isolation/BABEL-4146.spec:25]: added read write dependencies cases for serializable (serialisation anomaly)

File: test/JDBC/expected/BABEL-3214.out (hunk 13 above)
Comment (tanscorpio7) [reply test/JDBC/expected/BABEL-3214.out]: done

File: contrib/babelfishpg_tsql/src/guc.c (hunk 12 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/guc.c]: done

File: contrib/babelfishpg_tsql/src/guc.c (hunk 11 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/guc.c]: two gucs now

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y (hunk 10 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: done

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y (hunk 9 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: done

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y (hunk 8 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: done

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y (hunk 7 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: done

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 17]
```diff
@@ -4221,11 +4221,18 @@ tsql_IsolationLevelStr:
 				}
 			| REPEATABLE READ
 				{
-					TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
-					ereport(ERROR,
+					if(pltsql_enable_repeatable_read_isolation_level){
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: let us follow the coding convention as per file
if (x)
{

}

File: contrib/babelfishpg_tsql/src/guc.c [hunk 18]
```diff
@@ -1148,6 +1150,24 @@ define_custom_variables(void)
...
 							 GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE | GUC_DISALLOW_IN_AUTO_FILE,
 							 NULL, NULL, NULL);
+
+	DefineCustomBoolVariable("babelfishpg_tsql.enable_repeatable_read_isolation_level",
+							 gettext_noop("Enables support for isolation level reapeatable read"),
+							 NULL,
+							 &pltsql_enable_repeatable_read_isolation_level,
+							 false,
+							 PGC_USERSET,
+							 GUC_NOT_IN_SAMPLE ,
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c]: GUC_NOT_IN_SAMPLE ->  GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE | GUC_DISALLOW_IN_AUTO_FILE

**Review by surendravishnoi — COMMENTED** [review]
Please update commit message to only have change description (remove guidelines)

**Review by surendravishnoi — DISMISSED** [review]

File: test/dotnet/ExpectedOutput/TestIsolationLevel.out [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+on
+#Q#SELECT set_config('babelfishpg_tsql.enable_serializable_isolation_level','off',false)
+#D#text
+off
+#Q#select @@trancount;
+#D#int
+1
+#Q#select current_setting('transaction_isolation')
+#D#text
+read committed
```
Comment (surendravishnoi) [inline test/dotnet/ExpectedOutput/TestIsolationLevel.out]: why is this read committed, ideally both transaction and default isolation levels should be same. Could you please take a look?

File: test/dotnet/ExpectedOutput/TestIsolationLevel.out [hunk 20]
```diff
@@ -0,0 +1,48 @@
+#Q#SELECT set_config('babelfishpg_tsql.enable_repeatable_read_isolation_level','on',false)
```
Comment (surendravishnoi) [inline test/dotnet/ExpectedOutput/TestIsolationLevel.out]: We should add test for all 5 isolation levels.

File: test/dotnet/ExpectedOutput/TestIsolationLevel.out (hunk 19 above)
Comment (tanscorpio7) [reply test/dotnet/ExpectedOutput/TestIsolationLevel.out]: txn#!#begin#!#isolation#!#rr
This is equivalent to first begin transaction then set isolation level
hence we only have default_transaction_isolation set to rr

Comment (tanscorpio7) [reply test/dotnet/ExpectedOutput/TestIsolationLevel.out]: If i run set transaction isolation level reapeatable read
and then begin transaction 
both transaction_isolation & default are set to repeatable read

Comment (tanscorpio7) [reply test/dotnet/ExpectedOutput/TestIsolationLevel.out]: Fixed with BABEL-4411

File: contrib/babelfishpg_tsql/sql/sys_procedures.sql [hunk 21]
```diff
@@ -106,7 +106,9 @@ CREATE VIEW sys.babelfish_configurations_view as
     FROM pg_catalog.pg_settings 
     WHERE name collate "C" like 'babelfishpg_tsql.explain_%' OR
           name collate "C" like 'babelfishpg_tsql.escape_hatch_%' OR
-          name collate "C" = 'babelfishpg_tsql.enable_pg_hint';
+          name collate "C" = 'babelfishpg_tsql.enable_pg_hint' OR
+          name collate "C" = 'babelfishpg_tsql.repeatable_read_isolation' OR
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/sql/sys_procedures.sql]: let us call these isolation_* so that we can club them with single prefix

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 22]
```diff
@@ -4338,11 +4338,20 @@ tsql_IsolationLevelStr:
 				}
 			| REPEATABLE READ
 				{
-					TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
-					ereport(ERROR,
+					if (strcmp(pltsql_repeatable_read_isolation, "pg_isolation") == 0)
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: have we got feedback from PM about name? let us send a mail to ask with possible alternatives. pg_isolation looks good to me.

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 23]
```diff
@@ -4338,11 +4338,20 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
+						$$ = "repeatable read";	
+					}
+					else
+					{
+						TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
+						ereport(ERROR,
 							(errcode(ERRCODE_SYNTAX_ERROR),
-							errmsg("REPEATABLE READ isolation level is not supported"),
+							errmsg("Repeatable read is not supported by default, set configuration parameter 'repeatable_read_isolation' to 'pg_isolation' to get PG repeatable read"),
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: let us also get approval for error message from PM. It would be sufficient to just point out the config name without specifying which setting to use. Customer can get more details from documentation.

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 24]
```diff
@@ -4351,11 +4360,19 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
+						$$ = "serializable";
+					}
+					else
+					{
+						TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
+						ereport(ERROR,
 							(errcode(ERRCODE_SYNTAX_ERROR),
-							errmsg("SERIALIZABLE isolation level is not supported"),
+							errmsg("Serializable is not supported by default, set configuration parameter 'serializable_isolation' to 'pg_isolation' to get PG serializable"),
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: same here.

File: contrib/babelfishpg_tsql/src/guc.c [hunk 25]
```diff
@@ -805,6 +816,24 @@ define_custom_variables(void)
 							   PGC_USERSET,
 							   GUC_NOT_IN_SAMPLE | GUC_NO_SHOW_ALL | GUC_NO_RESET_ALL | GUC_DISALLOW_IN_FILE | GUC_DISALLOW_IN_AUTO_FILE,
 							   NULL, NULL, NULL);
+	
+	DefineCustomStringVariable("babelfishpg_tsql.repeatable_read_isolation",
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c]: I think it would be better to use ENUMs as possible values rather than strings.

File: test/JDBC/expected/BABEL_4145.out [hunk 26]
```diff
@@ -0,0 +1,308 @@
+-- tsql
+EXEC sp_babelfish_configure 'repeatable_read_isolation', 'default', 'server'
```
Comment (surendravishnoi) [inline test/JDBC/expected/BABEL_4145.out]: let us add a test for assigning invalid value to isolation level if not already

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 27]
```diff
@@ -4338,11 +4338,20 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
+						$$ = "repeatable read";	
+					}
+					else
+					{
+						TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_REPEATABLE_READ);
+						ereport(ERROR,
 							(errcode(ERRCODE_SYNTAX_ERROR),
-							errmsg("REPEATABLE READ isolation level is not supported"),
+							errmsg("‘REPEATABLE READ’ is not currently supported in Babelfish. Please use ‘babelfishpg_tsql.isolation_level_ repeatable_read’ to get PG repeatable read."),
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: Isolation level 'REPEATABLE READ' is not ...  to get PG repeatable read isolation level

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y [hunk 28]
```diff
@@ -4351,11 +4360,19 @@ tsql_IsolationLevelStr:
...
+						TSQLInstrumentation(INSTR_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
+						$$ = "serializable";
+					}
+					else
+					{
+						TSQLInstrumentation(INSTR_UNSUPPORTED_TSQL_ISOLATION_LEVEL_SERIALIZABLE);
+						ereport(ERROR,
 							(errcode(ERRCODE_SYNTAX_ERROR),
-							errmsg("SERIALIZABLE isolation level is not supported"),
+							errmsg("‘SERIALIZABLE’ is not currently supported in Babelfish. Please use ‘babelfishpg_tsql.isolation_level_serializable’ to get PG serializable."),
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y]: same here

File: contrib/babelfishpg_tsql/src/guc.c [hunk 29]
```diff
@@ -1207,6 +1212,8 @@ int			escape_hatch_rowversion = EH_STRICT;
 int			escape_hatch_showplan_all = EH_STRICT;
 int			escape_hatch_checkpoint = EH_IGNORE;
 int			escape_hatch_set_transaction_isolation_level = EH_STRICT;
+int			pltsql_isolation_level_repeatable_read = ISOLATION_OFF;
+int 		pltsql_isolation_level_serializable = ISOLATION_OFF;
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c:1217]: indent

File: contrib/babelfishpg_tsql/src/guc.c [hunk 30]
```diff
@@ -61,13 +61,12 @@ bool		pltsql_showplan_xml = false;
 bool		pltsql_fmtonly = false;
 bool		pltsql_enable_tsql_information_schema = true;
 bool		pltsql_no_browsetable = false;
+bool		pltsql_enable_create_alter_view_from_pg = false;
 
 char	   *pltsql_host_destribution = NULL;
 char	   *pltsql_host_release = NULL;
 char	   *pltsql_host_service_pack_level = NULL;
 
-bool		pltsql_enable_create_alter_view_from_pg = false;
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/src/guc.c]: why this change?

File: contrib/babelfishpg_tsql/src/guc.c (hunk 29 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/guc.c:1217]: Not sure why this it is off, both are 2 tabs & 1 space

**Review by surendravishnoi — DISMISSED** [review]

**Review by surendravishnoi — APPROVED** [review]
